File: projeto_biblioteca/models/nbibliotecario.js
const CRUD = require('./CRUD');
const fs = require('fs');

class Bibliotecario extends CRUD {
    constructor(nome, id) {
        super('bibliotecarios.json');
        this.nome = nome;
        this.id = id;
    }

    abrir() {
        return this.listar();
    }

    salvar() {
        const dados = this.listar();
        dados.push(this);
        fs.writeFileSync(this.caminhoArquivo, JSON.stringify(dados, null, 2));
    }
}

module.exports = Bibliotecario;

File: projeto_biblioteca/models/nbibliotecario.rb
require 'json'
require_relative 'crud'

class Bibliotecario < Crud
  attr_accessor :nome, :id

  def initialize(nome, id)
    super('dados/bibliotecarios.json')
    @nome = nome
    @id = id
  end

  def abrir
    listar
  end

  def salvar
    dados = listar
    dados << { nome: @nome, id: @id }
    super(dados)
  end
end
